### DISCUSSION TITLE
blog/2021-07-12-got-overdispersion-try-observation-level-random-effects-with-the-poisson-lognormal-mixture/

### DISCUSSION THREAD

**Discussion by giscus** [ASKurz/blogdown5#26] [Comments]
# blog/2021-07-12-got-overdispersion-try-observation-level-random-effects-with-the-poisson-lognormal-mixture/

It turns out that you can use random effects on cross-sectional count data. Yes, that's right. Each count gets its own random effect. Some people call this observation-level random effects and it can be a tricky way to handle overdispersion. The purpose of this post is to show how to do this and to try to make sense of what it even means.

https://solomonkurz.netlify.app/blog/2021-07-12-got-overdispersion-try-observation-level-random-effects-with-the-poisson-lognormal-mixture/

**Comment by jamesuanhoro** [comment]
Currently preparing a lecture on compounds & mixtures with counts, so decided to search for anyone using Poisson-log-normal with brms and LOOIC for comparison and I ended up here. Thanks for the example :).

One comment worth making is that the Poisson-gamma likelihood in brms is the marginal likelihood, so no gamma distribution observation level random effects (OLRE). While the PLN draws those OLREs. AFAIK, it makes the LOO comparison not so valid. To make the loo comparison valid, you'd likely want the PLN marginal likelihood. It's implemented in `poilog::dpoilog` function. My sense is the models would likely be less separable if you did this.

In my example, assuming `ln_model` is my PLN model and `y_var` is my count outcome, the code was:

```
dim(mu_draws <- fitted(
  ln_model,
  re_formula = ~ 0, scale = "linear", summary = FALSE
))
sigma_draws <- unname(posterior::as_draws_matrix(
  ln_model, "sd_case_id__Intercept"
)[, 1, drop = TRUE])
dim(ll_mat <- matrix(nrow = nrow(mu_draws), ncol = ncol(mu_draws)))
for (iter in seq_len(nrow(mu_draws))) {
  mu <- mu_draws[iter, , drop = TRUE]
  s <- sigma_draws[iter]
  for (j in seq_along(mu)) {
    ll_mat[iter, j] <- poilog::dpoilog(y_var[j], mu[j], s)
  }
}
loo_res <- loo::loo(log(ll_mat))
loo_res
```

I'm not as conversant with extracting draws out of brms fits, lol. And `dpoilog()` cannot take parameter vectors.

**Comment by ASKurz** [comment]
Erik Ringen has a nice blog post on pp-checks for OLRE Poisson models using the Oceanic toolkit data set. https://erikringen.github.io/posts/OLRE/
